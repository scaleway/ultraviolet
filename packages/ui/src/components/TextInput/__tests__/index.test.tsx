import { fireEvent, screen } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event'
import { renderWithTheme, shouldMatchEmotionSnapshot } from '@utils/test'
import { describe, expect, test, vi } from 'vitest'
import { TextInput } from '..'

describe('textInput', () => {
  test('should render correctly with basic props', () =>
    shouldMatchEmotionSnapshot(
      <TextInput label="Test" onChange={() => {}} value="test" />,
    ))

  test('should control the value', () => {
    const onChange = vi.fn()
    const onChangeValue = vi.fn()

    renderWithTheme(
      <TextInput
        label="Test"
        onChange={onChange}
        onChangeValue={onChangeValue}
        value="test"
      />,
    )

    const textarea = screen.getByLabelText<HTMLInputElement>('Test')
    expect(textarea.value).toBe('test')
    fireEvent.change(textarea, { target: { value: 'another value' } })
    expect(onChange).toHaveBeenCalled()
    expect(onChangeValue).toHaveBeenCalledWith('another value')
  })

  test('should be clearable', async () => {
    const onChange = vi.fn()

    renderWithTheme(
      <TextInput clearable label="Test" onChange={onChange} value="test" />,
    )

    const textarea = screen.getByLabelText<HTMLTextAreaElement>('Test')
    expect(textarea.value).toBe('test')
    const clearableButton = screen.getByLabelText('clear value')
    await userEvent.click(clearableButton)
    expect(onChange).toHaveBeenCalledWith({
      currentTarget: { value: '' },
      target: { value: '' },
    })
  })

  test('should render correctly when input is disabled', () =>
    shouldMatchEmotionSnapshot(
      <TextInput disabled label="Test" onChange={() => {}} value="test" />,
    ))

  test('should render correctly when input is readOnly', () =>
    shouldMatchEmotionSnapshot(
      <TextInput label="Test" onChange={() => {}} readOnly value="test" />,
    ))

  test('should render correctly when input has a success sentiment', () =>
    shouldMatchEmotionSnapshot(
      <TextInput
        label="Test"
        onChange={() => {}}
        success="success"
        value="test"
      />,
    ))

  test('should render correctly when input  has a error sentiment', () =>
    shouldMatchEmotionSnapshot(
      <TextInput
        error="success"
        label="Test"
        onChange={() => {}}
        value="test"
      />,
    ))

  test('should display success message', () => {
    const onChange = vi.fn()
    const successMessage = 'success message'

    renderWithTheme(
      <TextInput
        label="Test"
        onChange={onChange}
        success={successMessage}
        value="test"
      />,
    )

    expect(screen.getByText(successMessage)).toBeDefined()
  })

  test('should display error message', () => {
    const onChange = vi.fn()
    const errorMessage = 'error!'

    renderWithTheme(
      <TextInput
        label="Test"
        onChange={onChange}
        success={errorMessage}
        value="test"
      />,
    )

    expect(screen.getByText(errorMessage)).toBeDefined()
  })

  test('should display helper message', () => {
    const onChange = vi.fn()
    const helperMessage = 'helper'

    renderWithTheme(
      <TextInput
        helper={helperMessage}
        label="Test"
        onChange={onChange}
        value="test"
      />,
    )

    expect(screen.getByText(helperMessage)).toBeDefined()
  })

  test('should not display helper message when success is displayed', () => {
    const onChange = vi.fn()
    const successMessage = 'success message'
    const helperMessage = 'helper'

    renderWithTheme(
      <TextInput
        helper={helperMessage}
        label="Test"
        onChange={onChange}
        success={successMessage}
        value="test"
      />,
    )

    expect(screen.getByText(successMessage)).toBeDefined()
    expect(screen.queryByText(helperMessage)).toBeNull()
  })

  test('should not display helper message when error is displayed', () => {
    const onChange = vi.fn()
    const error = 'error!'
    const helperMessage = 'helper'

    renderWithTheme(
      <TextInput
        error={error}
        helper={helperMessage}
        label="Test"
        onChange={onChange}
        value="test"
      />,
    )

    expect(screen.getByText(error)).toBeDefined()
    expect(screen.queryByText(helperMessage)).toBeNull()
  })
})
