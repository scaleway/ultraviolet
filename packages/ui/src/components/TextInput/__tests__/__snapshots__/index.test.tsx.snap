// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`textInput > should render correctly when input  has a error sentiment 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 3rem;
  background: #ffffff;
  border: 1px solid #d9dadd;
  border-radius: 0.25rem;
}

.emotion-0>.emotion-3 {
  color: #3f4250;
}

.emotion-0>.emotion-3::-webkit-input-placeholder {
  color: #727683;
}

.emotion-0>.emotion-3::-moz-placeholder {
  color: #727683;
}

.emotion-0>.emotion-3:-ms-input-placeholder {
  color: #727683;
}

.emotion-0>.emotion-3::placeholder {
  color: #727683;
}

.emotion-0[data-success='true'] {
  border-color: #22674e;
}

.emotion-0[data-error='true'] {
  border-color: #b3144d;
}

.emotion-0[data-readonly='true'] {
  background: #f9f9fa;
  border-color: #d9dadd;
}

.emotion-0[data-disabled='true'] {
  background: #f3f3f4;
  border-color: #e9eaeb;
}

.emotion-0[data-disabled='true']>.emotion-3 {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3::-webkit-input-placeholder {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3::-moz-placeholder {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3:-ms-input-placeholder {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3::placeholder {
  color: #b5b7bd;
}

.emotion-0:not([data-disabled='true']):not([data-readonly]):hover {
  border-color: #8c40ef;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 3rem;
  background: #ffffff;
  border: 1px solid #d9dadd;
  border-radius: 0.25rem;
}

.emotion-0>.emotion-3 {
  color: #3f4250;
}

.emotion-0>.emotion-3::-webkit-input-placeholder {
  color: #727683;
}

.emotion-0>.emotion-3::-moz-placeholder {
  color: #727683;
}

.emotion-0>.emotion-3:-ms-input-placeholder {
  color: #727683;
}

.emotion-0>.emotion-3::placeholder {
  color: #727683;
}

.emotion-0[data-success='true'] {
  border-color: #22674e;
}

.emotion-0[data-error='true'] {
  border-color: #b3144d;
}

.emotion-0[data-readonly='true'] {
  background: #f9f9fa;
  border-color: #d9dadd;
}

.emotion-0[data-disabled='true'] {
  background: #f3f3f4;
  border-color: #e9eaeb;
}

.emotion-0[data-disabled='true']>.emotion-3 {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3::-webkit-input-placeholder {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3::-moz-placeholder {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3:-ms-input-placeholder {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3::placeholder {
  color: #b5b7bd;
}

.emotion-0:not([data-disabled='true']):not([data-readonly]):hover {
  border-color: #8c40ef;
}

.emotion-2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: none;
  outline: none;
  height: 100%;
  width: 100%;
  padding-left: 1rem;
  background: transparent;
  font-size: 0.875rem;
}

.emotion-2[data-size='large'] {
  font-size: 1rem;
}

.emotion-2[data-size='small'] {
  padding-left: 0.5rem;
}

.emotion-2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: none;
  outline: none;
  height: 100%;
  width: 100%;
  padding-left: 1rem;
  background: transparent;
  font-size: 0.875rem;
}

.emotion-2[data-size='large'] {
  font-size: 1rem;
}

.emotion-2[data-size='small'] {
  padding-left: 0.5rem;
}

.emotion-4 {
  padding: 0 1rem;
}

.emotion-4 {
  padding: 0 1rem;
}

<div
    data-testid="testing"
  >
    <div
      class="styles__toi52u0 styles_alignItems_normal_xxsmall__toi52u31 styles_flexDirection_column_xxsmall__toi52u2d styles_flexWrap_nowrap_xxsmall__toi52u7d styles_gap_0.25rem_xxsmall__toi52u1v styles_justifyContent_normal_xxsmall__toi52u5d"
    >
      <label
        class="styles_htmlFor__17td3g91 style__m4c9ow0 style_strikeThrough_false__m4c9ow2 style_italic_false__m4c9ow4 style_underline_false__m4c9ow6 style_oneLine_false__m4c9ow8 style_sentiment_neutral__m4c9owb style_prominence_default__m4c9owi style_variant_bodyStrong__m4c9owq style_disabled_false__m4c9ow1b style_undefined_compound_24__m4c9ow20 style_undefined_compound_72__m4c9ow3c"
        for="«rk»"
      >
        Test
      </label>
      <div>
        <div
          class="emotion-0 emotion-1"
          data-disabled="false"
          data-error="true"
          data-readonly="false"
          data-success="false"
        >
          <input
            aria-invalid="true"
            class="emotion-2 emotion-3"
            data-size="large"
            id="«rk»"
            type="text"
            value="test"
          />
          <div
            class="emotion-4 emotion-5 styles__toi52u0 styles_alignItems_center_xxsmall__toi52u3d styles_flexDirection_row_xxsmall__toi52u2j styles_flexWrap_nowrap_xxsmall__toi52u7d styles_gap_0.5rem_xxsmall__toi52u7 styles_justifyContent_normal_xxsmall__toi52u5d"
          >
            <svg
              class="styles__vbm0wq0 styles_prominence_default__vbm0wqg styles_disabled_false__vbm0wqf styles_sentiment_danger__vbm0wq5 styles_size_small__vbm0wqa styles_undefined_compound_36__vbm0wq1k"
              viewBox="0 0 16 16"
            >
              <path
                clip-rule="evenodd"
                d="M8 15C11.866 15 15 11.866 15 8C15 4.13401 11.866 1 8 1C4.13401 1 1 4.13401 1 8C1 11.866 4.13401 15 8 15ZM8 4C8.41421 4 8.75 4.33579 8.75 4.75V7.75C8.75 8.16421 8.41421 8.5 8 8.5C7.58579 8.5 7.25 8.16421 7.25 7.75V4.75C7.25 4.33579 7.58579 4 8 4ZM8 12C8.55228 12 9 11.5523 9 11C9 10.4477 8.55228 10 8 10C7.44772 10 7 10.4477 7 11C7 11.5523 7.44772 12 8 12Z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
      </div>
      <p
        class="style__m4c9ow0 style_strikeThrough_false__m4c9ow2 style_italic_false__m4c9ow4 style_underline_false__m4c9ow6 style_oneLine_false__m4c9ow8 style_sentiment_danger__m4c9owd style_prominence_default__m4c9owi style_variant_caption__m4c9ows style_disabled_false__m4c9ow1b style_undefined_compound_8__m4c9ow1k style_undefined_compound_72__m4c9ow3c"
      >
        success
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`textInput > should render correctly when input has a success sentiment 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 3rem;
  background: #ffffff;
  border: 1px solid #d9dadd;
  border-radius: 0.25rem;
}

.emotion-0>.emotion-3 {
  color: #3f4250;
}

.emotion-0>.emotion-3::-webkit-input-placeholder {
  color: #727683;
}

.emotion-0>.emotion-3::-moz-placeholder {
  color: #727683;
}

.emotion-0>.emotion-3:-ms-input-placeholder {
  color: #727683;
}

.emotion-0>.emotion-3::placeholder {
  color: #727683;
}

.emotion-0[data-success='true'] {
  border-color: #22674e;
}

.emotion-0[data-error='true'] {
  border-color: #b3144d;
}

.emotion-0[data-readonly='true'] {
  background: #f9f9fa;
  border-color: #d9dadd;
}

.emotion-0[data-disabled='true'] {
  background: #f3f3f4;
  border-color: #e9eaeb;
}

.emotion-0[data-disabled='true']>.emotion-3 {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3::-webkit-input-placeholder {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3::-moz-placeholder {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3:-ms-input-placeholder {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3::placeholder {
  color: #b5b7bd;
}

.emotion-0:not([data-disabled='true']):not([data-readonly]):hover {
  border-color: #8c40ef;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 3rem;
  background: #ffffff;
  border: 1px solid #d9dadd;
  border-radius: 0.25rem;
}

.emotion-0>.emotion-3 {
  color: #3f4250;
}

.emotion-0>.emotion-3::-webkit-input-placeholder {
  color: #727683;
}

.emotion-0>.emotion-3::-moz-placeholder {
  color: #727683;
}

.emotion-0>.emotion-3:-ms-input-placeholder {
  color: #727683;
}

.emotion-0>.emotion-3::placeholder {
  color: #727683;
}

.emotion-0[data-success='true'] {
  border-color: #22674e;
}

.emotion-0[data-error='true'] {
  border-color: #b3144d;
}

.emotion-0[data-readonly='true'] {
  background: #f9f9fa;
  border-color: #d9dadd;
}

.emotion-0[data-disabled='true'] {
  background: #f3f3f4;
  border-color: #e9eaeb;
}

.emotion-0[data-disabled='true']>.emotion-3 {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3::-webkit-input-placeholder {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3::-moz-placeholder {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3:-ms-input-placeholder {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3::placeholder {
  color: #b5b7bd;
}

.emotion-0:not([data-disabled='true']):not([data-readonly]):hover {
  border-color: #8c40ef;
}

.emotion-2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: none;
  outline: none;
  height: 100%;
  width: 100%;
  padding-left: 1rem;
  background: transparent;
  font-size: 0.875rem;
}

.emotion-2[data-size='large'] {
  font-size: 1rem;
}

.emotion-2[data-size='small'] {
  padding-left: 0.5rem;
}

.emotion-2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: none;
  outline: none;
  height: 100%;
  width: 100%;
  padding-left: 1rem;
  background: transparent;
  font-size: 0.875rem;
}

.emotion-2[data-size='large'] {
  font-size: 1rem;
}

.emotion-2[data-size='small'] {
  padding-left: 0.5rem;
}

.emotion-4 {
  padding: 0 1rem;
}

.emotion-4 {
  padding: 0 1rem;
}

<div
    data-testid="testing"
  >
    <div
      class="styles__toi52u0 styles_alignItems_normal_xxsmall__toi52u31 styles_flexDirection_column_xxsmall__toi52u2d styles_flexWrap_nowrap_xxsmall__toi52u7d styles_gap_0.25rem_xxsmall__toi52u1v styles_justifyContent_normal_xxsmall__toi52u5d"
    >
      <label
        class="styles_htmlFor__17td3g91 style__m4c9ow0 style_strikeThrough_false__m4c9ow2 style_italic_false__m4c9ow4 style_underline_false__m4c9ow6 style_oneLine_false__m4c9ow8 style_sentiment_neutral__m4c9owb style_prominence_default__m4c9owi style_variant_bodyStrong__m4c9owq style_disabled_false__m4c9ow1b style_undefined_compound_24__m4c9ow20 style_undefined_compound_72__m4c9ow3c"
        for="«rg»"
      >
        Test
      </label>
      <div>
        <div
          class="emotion-0 emotion-1"
          data-disabled="false"
          data-error="false"
          data-readonly="false"
          data-success="true"
        >
          <input
            aria-invalid="false"
            class="emotion-2 emotion-3"
            data-size="large"
            id="«rg»"
            type="text"
            value="test"
          />
          <div
            class="emotion-4 emotion-5 styles__toi52u0 styles_alignItems_center_xxsmall__toi52u3d styles_flexDirection_row_xxsmall__toi52u2j styles_flexWrap_nowrap_xxsmall__toi52u7d styles_gap_0.5rem_xxsmall__toi52u7 styles_justifyContent_normal_xxsmall__toi52u5d"
          >
            <svg
              class="styles__vbm0wq0 styles_prominence_default__vbm0wqg styles_disabled_false__vbm0wqf styles_sentiment_success__vbm0wq4 styles_size_small__vbm0wqa styles_undefined_compound_35__vbm0wq1j"
              viewBox="0 0 16 16"
            >
              <path
                clip-rule="evenodd"
                d="M8 15C11.866 15 15 11.866 15 8C15 4.13401 11.866 1 8 1C4.13401 1 1 4.13401 1 8C1 11.866 4.13401 15 8 15ZM11.8435 6.20859C12.0967 5.88082 12.0363 5.40981 11.7086 5.15654C11.3808 4.90327 10.9098 4.96365 10.6565 5.29141L6.95615 10.0801L5.30747 8.24828C5.03038 7.94039 4.55616 7.91543 4.24828 8.19253C3.94039 8.46962 3.91544 8.94384 4.19253 9.25172L6.44253 11.7517C6.59132 11.917 6.80582 12.0078 7.02809 11.9995C7.25036 11.9911 7.45746 11.8846 7.59346 11.7086L11.8435 6.20859Z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
      </div>
      <p
        class="style__m4c9ow0 style_strikeThrough_false__m4c9ow2 style_italic_false__m4c9ow4 style_underline_false__m4c9ow6 style_oneLine_false__m4c9ow8 style_sentiment_success__m4c9owc style_prominence_default__m4c9owi style_variant_caption__m4c9ows style_disabled_false__m4c9ow1b style_undefined_compound_16__m4c9ow1s style_undefined_compound_72__m4c9ow3c"
      >
        success
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`textInput > should render correctly when input is disabled 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 3rem;
  background: #ffffff;
  border: 1px solid #d9dadd;
  border-radius: 0.25rem;
}

.emotion-0>.emotion-3 {
  color: #3f4250;
}

.emotion-0>.emotion-3::-webkit-input-placeholder {
  color: #727683;
}

.emotion-0>.emotion-3::-moz-placeholder {
  color: #727683;
}

.emotion-0>.emotion-3:-ms-input-placeholder {
  color: #727683;
}

.emotion-0>.emotion-3::placeholder {
  color: #727683;
}

.emotion-0[data-success='true'] {
  border-color: #22674e;
}

.emotion-0[data-error='true'] {
  border-color: #b3144d;
}

.emotion-0[data-readonly='true'] {
  background: #f9f9fa;
  border-color: #d9dadd;
}

.emotion-0[data-disabled='true'] {
  background: #f3f3f4;
  border-color: #e9eaeb;
}

.emotion-0[data-disabled='true']>.emotion-3 {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3::-webkit-input-placeholder {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3::-moz-placeholder {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3:-ms-input-placeholder {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3::placeholder {
  color: #b5b7bd;
}

.emotion-0:not([data-disabled='true']):not([data-readonly]):hover {
  border-color: #8c40ef;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 3rem;
  background: #ffffff;
  border: 1px solid #d9dadd;
  border-radius: 0.25rem;
}

.emotion-0>.emotion-3 {
  color: #3f4250;
}

.emotion-0>.emotion-3::-webkit-input-placeholder {
  color: #727683;
}

.emotion-0>.emotion-3::-moz-placeholder {
  color: #727683;
}

.emotion-0>.emotion-3:-ms-input-placeholder {
  color: #727683;
}

.emotion-0>.emotion-3::placeholder {
  color: #727683;
}

.emotion-0[data-success='true'] {
  border-color: #22674e;
}

.emotion-0[data-error='true'] {
  border-color: #b3144d;
}

.emotion-0[data-readonly='true'] {
  background: #f9f9fa;
  border-color: #d9dadd;
}

.emotion-0[data-disabled='true'] {
  background: #f3f3f4;
  border-color: #e9eaeb;
}

.emotion-0[data-disabled='true']>.emotion-3 {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3::-webkit-input-placeholder {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3::-moz-placeholder {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3:-ms-input-placeholder {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3::placeholder {
  color: #b5b7bd;
}

.emotion-0:not([data-disabled='true']):not([data-readonly]):hover {
  border-color: #8c40ef;
}

.emotion-2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: none;
  outline: none;
  height: 100%;
  width: 100%;
  padding-left: 1rem;
  background: transparent;
  font-size: 0.875rem;
}

.emotion-2[data-size='large'] {
  font-size: 1rem;
}

.emotion-2[data-size='small'] {
  padding-left: 0.5rem;
}

.emotion-2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: none;
  outline: none;
  height: 100%;
  width: 100%;
  padding-left: 1rem;
  background: transparent;
  font-size: 0.875rem;
}

.emotion-2[data-size='large'] {
  font-size: 1rem;
}

.emotion-2[data-size='small'] {
  padding-left: 0.5rem;
}

<div
    data-testid="testing"
  >
    <div
      class="styles__toi52u0 styles_alignItems_normal_xxsmall__toi52u31 styles_flexDirection_column_xxsmall__toi52u2d styles_flexWrap_nowrap_xxsmall__toi52u7d styles_gap_0.25rem_xxsmall__toi52u1v styles_justifyContent_normal_xxsmall__toi52u5d"
    >
      <label
        class="styles_htmlFor__17td3g91 style__m4c9ow0 style_strikeThrough_false__m4c9ow2 style_italic_false__m4c9ow4 style_underline_false__m4c9ow6 style_oneLine_false__m4c9ow8 style_sentiment_neutral__m4c9owb style_prominence_default__m4c9owi style_variant_bodyStrong__m4c9owq style_disabled_false__m4c9ow1b style_undefined_compound_24__m4c9ow20 style_undefined_compound_72__m4c9ow3c"
        for="«ra»"
      >
        Test
      </label>
      <div>
        <div
          class="emotion-0 emotion-1"
          data-disabled="true"
          data-error="false"
          data-readonly="false"
          data-success="false"
        >
          <input
            aria-invalid="false"
            class="emotion-2 emotion-3"
            data-size="large"
            disabled=""
            id="«ra»"
            type="text"
            value="test"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`textInput > should render correctly when input is readOnly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 3rem;
  background: #ffffff;
  border: 1px solid #d9dadd;
  border-radius: 0.25rem;
}

.emotion-0>.emotion-3 {
  color: #3f4250;
}

.emotion-0>.emotion-3::-webkit-input-placeholder {
  color: #727683;
}

.emotion-0>.emotion-3::-moz-placeholder {
  color: #727683;
}

.emotion-0>.emotion-3:-ms-input-placeholder {
  color: #727683;
}

.emotion-0>.emotion-3::placeholder {
  color: #727683;
}

.emotion-0[data-success='true'] {
  border-color: #22674e;
}

.emotion-0[data-error='true'] {
  border-color: #b3144d;
}

.emotion-0[data-readonly='true'] {
  background: #f9f9fa;
  border-color: #d9dadd;
}

.emotion-0[data-disabled='true'] {
  background: #f3f3f4;
  border-color: #e9eaeb;
}

.emotion-0[data-disabled='true']>.emotion-3 {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3::-webkit-input-placeholder {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3::-moz-placeholder {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3:-ms-input-placeholder {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3::placeholder {
  color: #b5b7bd;
}

.emotion-0:not([data-disabled='true']):not([data-readonly]):hover {
  border-color: #8c40ef;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 3rem;
  background: #ffffff;
  border: 1px solid #d9dadd;
  border-radius: 0.25rem;
}

.emotion-0>.emotion-3 {
  color: #3f4250;
}

.emotion-0>.emotion-3::-webkit-input-placeholder {
  color: #727683;
}

.emotion-0>.emotion-3::-moz-placeholder {
  color: #727683;
}

.emotion-0>.emotion-3:-ms-input-placeholder {
  color: #727683;
}

.emotion-0>.emotion-3::placeholder {
  color: #727683;
}

.emotion-0[data-success='true'] {
  border-color: #22674e;
}

.emotion-0[data-error='true'] {
  border-color: #b3144d;
}

.emotion-0[data-readonly='true'] {
  background: #f9f9fa;
  border-color: #d9dadd;
}

.emotion-0[data-disabled='true'] {
  background: #f3f3f4;
  border-color: #e9eaeb;
}

.emotion-0[data-disabled='true']>.emotion-3 {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3::-webkit-input-placeholder {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3::-moz-placeholder {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3:-ms-input-placeholder {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3::placeholder {
  color: #b5b7bd;
}

.emotion-0:not([data-disabled='true']):not([data-readonly]):hover {
  border-color: #8c40ef;
}

.emotion-2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: none;
  outline: none;
  height: 100%;
  width: 100%;
  padding-left: 1rem;
  background: transparent;
  font-size: 0.875rem;
}

.emotion-2[data-size='large'] {
  font-size: 1rem;
}

.emotion-2[data-size='small'] {
  padding-left: 0.5rem;
}

.emotion-2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: none;
  outline: none;
  height: 100%;
  width: 100%;
  padding-left: 1rem;
  background: transparent;
  font-size: 0.875rem;
}

.emotion-2[data-size='large'] {
  font-size: 1rem;
}

.emotion-2[data-size='small'] {
  padding-left: 0.5rem;
}

<div
    data-testid="testing"
  >
    <div
      class="styles__toi52u0 styles_alignItems_normal_xxsmall__toi52u31 styles_flexDirection_column_xxsmall__toi52u2d styles_flexWrap_nowrap_xxsmall__toi52u7d styles_gap_0.25rem_xxsmall__toi52u1v styles_justifyContent_normal_xxsmall__toi52u5d"
    >
      <label
        class="styles_htmlFor__17td3g91 style__m4c9ow0 style_strikeThrough_false__m4c9ow2 style_italic_false__m4c9ow4 style_underline_false__m4c9ow6 style_oneLine_false__m4c9ow8 style_sentiment_neutral__m4c9owb style_prominence_default__m4c9owi style_variant_bodyStrong__m4c9owq style_disabled_false__m4c9ow1b style_undefined_compound_24__m4c9ow20 style_undefined_compound_72__m4c9ow3c"
        for="«rd»"
      >
        Test
      </label>
      <div>
        <div
          class="emotion-0 emotion-1"
          data-disabled="false"
          data-error="false"
          data-readonly="true"
          data-success="false"
        >
          <input
            aria-invalid="false"
            class="emotion-2 emotion-3"
            data-size="large"
            id="«rd»"
            readonly=""
            type="text"
            value="test"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`textInput > should render correctly with basic props 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 3rem;
  background: #ffffff;
  border: 1px solid #d9dadd;
  border-radius: 0.25rem;
}

.emotion-0>.emotion-3 {
  color: #3f4250;
}

.emotion-0>.emotion-3::-webkit-input-placeholder {
  color: #727683;
}

.emotion-0>.emotion-3::-moz-placeholder {
  color: #727683;
}

.emotion-0>.emotion-3:-ms-input-placeholder {
  color: #727683;
}

.emotion-0>.emotion-3::placeholder {
  color: #727683;
}

.emotion-0[data-success='true'] {
  border-color: #22674e;
}

.emotion-0[data-error='true'] {
  border-color: #b3144d;
}

.emotion-0[data-readonly='true'] {
  background: #f9f9fa;
  border-color: #d9dadd;
}

.emotion-0[data-disabled='true'] {
  background: #f3f3f4;
  border-color: #e9eaeb;
}

.emotion-0[data-disabled='true']>.emotion-3 {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3::-webkit-input-placeholder {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3::-moz-placeholder {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3:-ms-input-placeholder {
  color: #b5b7bd;
}

.emotion-0[data-disabled='true']>.emotion-3::placeholder {
  color: #b5b7bd;
}

.emotion-0:not([data-disabled='true']):not([data-readonly]):hover {
  border-color: #8c40ef;
}

.emotion-2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: none;
  outline: none;
  height: 100%;
  width: 100%;
  padding-left: 1rem;
  background: transparent;
  font-size: 0.875rem;
}

.emotion-2[data-size='large'] {
  font-size: 1rem;
}

.emotion-2[data-size='small'] {
  padding-left: 0.5rem;
}

<div
    data-testid="testing"
  >
    <div
      class="styles__toi52u0 styles_alignItems_normal_xxsmall__toi52u31 styles_flexDirection_column_xxsmall__toi52u2d styles_flexWrap_nowrap_xxsmall__toi52u7d styles_gap_0.25rem_xxsmall__toi52u1v styles_justifyContent_normal_xxsmall__toi52u5d"
    >
      <label
        class="styles_htmlFor__17td3g91 style__m4c9ow0 style_strikeThrough_false__m4c9ow2 style_italic_false__m4c9ow4 style_underline_false__m4c9ow6 style_oneLine_false__m4c9ow8 style_sentiment_neutral__m4c9owb style_prominence_default__m4c9owi style_variant_bodyStrong__m4c9owq style_disabled_false__m4c9ow1b style_undefined_compound_24__m4c9ow20 style_undefined_compound_72__m4c9ow3c"
        for="«r0»"
      >
        Test
      </label>
      <div>
        <div
          class="emotion-0 emotion-1"
          data-disabled="false"
          data-error="false"
          data-readonly="false"
          data-success="false"
        >
          <input
            aria-invalid="false"
            class="emotion-2 emotion-3"
            data-size="large"
            id="«r0»"
            type="text"
            value="test"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
